dotCMS/src/main/java/com/dotcms/cms/login/LoginServiceAPIFactory.java

```diff
@@ -699,6 +702,57 @@ public User getLoggedInUser( ){
         }
     }
 
+    /**
+     * This function notifies API Tokens that are about to expire within the next 7 days.
+     * If the user is an admin, it checks all users in the system, and notifies about all of them.
+     * If the user is not an admin, it checks and notifies only the logged in user.
+     */
+    private void messageTokensToExpire(User logInUser) throws DotDataException {
+
+        SystemMessageBuilder message = null;
+        ApiTokenAPI apiToken = APILocator.getApiTokenAPI();
+        if (logInUser.isAdmin()) {
+            List<User> users = APILocator.getUserAPI().findAllUsers();
+            for (User user : users) {
+                List<ApiToken> tokens = apiToken.findApiTokensByUserId(user.getUserId(), false, logInUser);
+                if (UtilMethods.isSet(tokens)) {
+                    if (apiToken.hasAnyTokenExpiring(tokens)) {
+                         message = new SystemMessageBuilder()
+                                .setMessage("User " + user.getFullName() + " has API Tokens that are about to expire within the next 7 days. Please review their tokens.")
+                                .setSeverity(MessageSeverity.WARNING)
+                                .setType(MessageType.SIMPLE_MESSAGE)
+                                .setLife(86400000);
+                         sendMessageDelayed(message, logInUser);
```

Comment:
I would sleep here a few milliseconds in case the client may have many users on db